### ISSUE TITLE
Dataset for training llava

### ISSUE THREAD

**Issue by LehongWu** [sled-group/RACER#8]
Hi! Two quick questions regarding training the high-level VLM (the LLaVA model):

1. In Sec. IV of the RACER paper, you mentioned two datasets: the original RLBench and your augmented one. But in the training script, it seems only the augmented data is used. Just to clarify—are both datasets used for training LLaVA?

2. The paper states that the augmented dataset has 5× the trajectories of the original dataset. However, the augmented dataset on HuggingFace (sled-umich/RACER-augmented_rlbench) appears smaller than the original dataset on Google Drive, and it only contains selected frames rather than all frames. Could you clarify what exact data were used?

**Comment by YinpeiDai** [comment]
Thanks for your interest in our work.

1. Only augmented datasets are used to train LLaVA. The original dataset does not contain language instructions, so can not be used to train.  

2. We use RVT as action backbone, which predict action at sparse key-frames, so we only need to augment selected frames to train both LLaVA and RVT.  

Hope that helps!

**Comment by LehongWu** [comment]
That helps a lot. Thanks!

**Comment by LehongWu** [comment]
Hi, in your huggingface data, I can see the "actions.csv" files, e.g. "augmented_rlbench/train/close_jar/0/actions.csv"
Basically I want to get the ground truth low-level action corresponding to each VLM instruction.
I get that the number of actions equals the number of images, but I am not sure how they are mapped. Can you explain a bit? Thanks so much! @YinpeiDai

**Comment by YinpeiDai** [comment]
Hi, the original actions for model training are stored inside the `obs.pkl` files, which follows `rlbench.backend.observation.Observation` with `gripper_pose`,   and `ignore_collisions` and `gripper_open` indicating the RLbench actions.  

The `actions.csv` is a flattened version of all actions (both failed and correct actions), each row is a list of 9 numbers (7DoF EE pose, gripper, collision).  But we did not use this for training, it's just for human reading to check some data generation issue.  When we developed it, we didn't label it with the index for better reading. In general, the order should go like this
`exper1 -> expert2 -> perturb 3-1 -> perturb 3-2 ...-> expert 3 ->  expert 4 ->  perturb 5-1 ->  intermediate 5-1 -> perturb 5-2 ->  intermediate 5-2 -> ... -> expert 5 -> expert 6 -> ...`
suppose expert 3 is a key alignment step to augment, expert 5 is a key grasping/releasing step to augment.

You can refer to this [data generation repo]( https://github.com/rich-language-failure-recovery/RACER-DataGen?tab=readme-ov-file#data-format) for more information

**Comment by YinpeiDai** [comment]
I'll close the issue for now, feel free to reopen it.
